plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaPageSink.java

```diff
@@ -52,14 +53,16 @@ public KafkaPageSink(
             List<KafkaColumnHandle> columns,
             RowEncoder keyEncoder,
             RowEncoder messageEncoder,
-            KafkaProducerFactory producerFactory)
+            KafkaProducerFactory producerFactory,
+            ConnectorSession session)
     {
         this.topicName = requireNonNull(topicName, "topicName is null");
         this.columns = requireNonNull(ImmutableList.copyOf(columns), "columns is null");
         this.keyEncoder = requireNonNull(keyEncoder, "keyEncoder is null");
         this.messageEncoder = requireNonNull(messageEncoder, "messageEncoder is null");
         requireNonNull(producerFactory, "producerFactory is null");
-        this.producer = producerFactory.create();
+        requireNonNull(session, "session is null");
+        this.producer = producerFactory.create(session);
```

Comment:
Can we move this to `KafkaPageSinkProvider` so we don't have to pass the ConnectorSession here

Reply:
I don't think this is a good idea, bacause `KafkaPageSink` is responsible for closing `producer`. I would prefer not to move ownership if it is not needed.